inst/include/mode/solver.hpp

```diff
@@ -190,20 +194,20 @@ class solver {
 
   std::vector<double>::iterator
   state(const std::vector<size_t>& index,
-             std::vector<double>::iterator end_state) const {
-    auto y = stepper_.output();
+             std::vector<double>::iterator end_state) {
+    auto it = state_full_.begin();
+    stepper_.state(it);
+    m_.output(t_, it);
```

Comment:
We can move to only the stepper holding `m_` which would be preferable - so this definitely needs moving deeper